### PR TITLE
Update for Feature Flag Examples

### PR DESCRIPTION
**Pull request by okbel**
### Update for Feature Flag Apple Store Example

- This PR updates the Apple Store example to to the latest API using middleware and edge functions.
- Additionally, this PR adds more functionality to showcase the power of doing Feature Flag at the Edge.
- Updates ENV vars to match the ones provided by Upstash.

### Demo URL

Preview URL

## How to Test
- Go to the index, check the store
- Close the store going to `/api/store/close`
- Check the index
- Open the store again `/api/store/open`
- DONE! 

### Type of Change

- [X] New Example
- [X] Example updates (Bug fixes, new features, etc.)
- [ ] Other (changes to the codebase, but not to examples)

### CODE REVIEW

File: edge-functions/feature-flag-apple-store/package.json [hunk 1]
```diff
@@ -4,24 +4,24 @@
...
-    "lint": "next lint"
+    "dev": "next dev",
+    "lint": "next lint",
+    "start": "next start"
   },
   "dependencies": {
-    "next": "^12.1.0",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "next": "12.1.7-canary.45",
```
Comment (lfades) [inline edge-functions/feature-flag-apple-store/package.json:13]: Lets use `canary` here as it's constantly getting updates, until we can move to `latest`:
```suggestion
    "next": "canary",
```

File: edge-functions/feature-flag-apple-store/package.json [hunk 2]
```diff
@@ -4,24 +4,24 @@
...
-    "eslint": "^8.9.0",
-    "eslint-config-next": "^12.1.0",
-    "postcss": "^8.4.6",
-    "tailwindcss": "^3.0.18",
-    "typescript": "^4.5.5"
+    "@types/node": "^18.0.0",
+    "@types/react": "latest",
+    "autoprefixer": "^10.4.7",
+    "eslint": "^8.18.0",
+    "eslint-config-next": "latest",
```
Comment (lfades) [inline edge-functions/feature-flag-apple-store/package.json:22]:
```suggestion
    "eslint-config-next": "canary",
```

File: edge-functions/feature-flag-apple-store/pages/api/store/close.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+export const config = {
+  runtime: 'experimental-edge',
+}
+
+export default async function CloseStore() {
+  try {
+    const { result } = await set('store-closed', 'true')
+
+    if (result !== 'OK') {
+      throw `Upstash throwed an error: ${result} `
```
Comment (lfades) [inline edge-functions/feature-flag-apple-store/pages/api/store/close.ts:12]: This is not an error as the call to upstash worked and was correct, but something else failed if the result is not `'OK'`
```suggestion
      throw new Error(`Unexpected result from Upstash: ${result}`)
```

File: edge-functions/feature-flag-apple-store/pages/api/store/open.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+export const config = {
+  runtime: 'experimental-edge',
+}
+
+export default async function OpenStore() {
+  try {
+    const { result } = await set('store-closed', 'false')
+
+    if (result !== 'OK') {
+      throw `Upstash throwed an error: ${result} `
```
Comment (lfades) [inline edge-functions/feature-flag-apple-store/pages/api/store/open.ts:12]: This is not an error as the call to upstash worked and was correct, but something else failed if the result is not `'OK'`
```suggestion
      throw new Error(`Unexpected result from Upstash: ${result}`)
```

**Review by lfades — CHANGES_REQUESTED** [review]
🕵️

File: edge-functions/feature-flag-apple-store/.gitignore [hunk 5]
```diff
@@ -25,6 +25,7 @@ yarn-debug.log*
 yarn-error.log*
 
 # Local ENV files
+.env
```
[the remark this answers is not in the document]
Comment (okbel) [reply edge-functions/feature-flag-apple-store/.gitignore:28]: `.env` should be there by convention. I added it to all examples but for some reason it was deleted.

Reason:

`vercel env pull` creates an .env. Doing a git add -A could lead to secrets leak.

File: edge-functions/feature-flag-apple-store/package.json (hunk 1 above)
Comment (okbel) [reply edge-functions/feature-flag-apple-store/package.json:13]: https://github.com/vercel/examples/pull/318#discussion_r905162311 Check this thread

**Review by lfades — APPROVED** [review]

**Pull request merged**